From 558d0b9e7566c232c564a67e635b8589ab8cf052 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Rodr=C3=ADguez?= <electronlibre88@gmail.com>
Date: Mon, 13 Feb 2012 20:45:42 +0100
Subject: [PATCH 1/9] README editado

---
 README |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/README b/README
index e69de29..e1657dd 100644
--- a/README
+++ b/README
@@ -0,0 +1 @@
+En este repositorio subiré pequeños scripts que vaya haciendo y colgando en el blog (elelectronlibre.wordpress.com) Saludos!
\ No newline at end of file
-- 
1.7.5.4

